### DISCUSSION TITLE
How are Vehicles spawned in Eclipse MOSAIC?

### DISCUSSION THREAD

**Discussion by greyweb** [eclipse/mosaic#180] [Q&A] [answered]
Hi all, I've been exploring the Barnim scenario and wanted to experiment the scenario , somwhere near my locality
For which:
1. I changed the RSU location by changing the Lat and Lon values in mapping.json 
2. Changed the region of Icy roads near the RSU in the environment.json file.

After doing this my RSU was correctly placed in my locality,But however the vehicles spawned from another random region(not from Barnim).

Also would like to understand better how lane and path variables can be set for custom locations.

Thanks in Advance! :)

**Comment by kschrab** [comment, marked as answer]
The spawn location should not be affected by moving the RSU or changing the event location. Vehicles follow a specific route   and are spawned at the start location of their route. The routes are stored in the `application/Barnim.db` file (SQLite database). Routes are identified by its ID and can be changed/added using the scenario-convert tool (find more details [here](https://www.eclipse.org/mosaic/docs/scenarios/create_a_new_scenario/#generating-routes)). The vehicles are spawned according to the traffic definition in the `mapping_config.json`. Here you can define one or more so-called "spawners", which will add a continuous flow of vehicles following a specific route:

The following configuration will spawn 120 vehicles of type `Car` on a route with ID `1`. The first vehicle will spawn 5 seconds after the simulation started. Subsequent vehicles will be spawned in a fixed time interval derived from the `targetFlow` (veh/h). The value of 1800 veh/h will spawn vehicles in a time interval of 2 seconds. The vehicles will spawn on lane 0 and 1 alternately.

```
vehicles": [
        {
            "startingTime": 5.0,
            "targetFlow": 1800,
            "maxNumberVehicles": 120,
            "route": "1",
            "lanes": [ 0, 1 ]
            "types": [ "Car" ]
        }
]
```

**Reply by greyweb** [reply]
Thanks @kschrab for the detailed reply. Will try out as you suggested. :)
